File: Python/Array/E-1431. Kids With the Greatest Number of Candies.py
class Solution:
    @classmethod
    def kidsWithCandies(self, candies: list[int], extraCandies: int) -> list[bool]:
        max_candies = max(candies)
        result = [True if c + extraCandies >= max_candies else False for c in candies]
        return result


testData = [
    {
        "candies": [2, 3, 5, 1, 3],
        "extraCandies": 3,
        "ans": [True, True, True, False, True],
    },
]

for data in testData:
    print(
        data["ans"] == Solution.kidsWithCandies(data["candies"], data["extraCandies"])
    )

File: Array/E-1431. Kids With the Greatest Number of Candies/E-1431. Kids With the Greatest Number of Candies.ts
function kidsWithCandies(candies: number[], extraCandies: number): boolean[] {
  const maxCandies = Math.max(...candies);
  return candies.map((item) => item + extraCandies >= maxCandies);
}

const testData = [
  {
    candies: [2, 3, 5, 1, 3],
    extraCandies: 3,
    ans: [true, true, true, false, true],
  },
];

for (const data of testData) {
  console.log(
    JSON.stringify(data.ans) ===
      JSON.stringify(kidsWithCandies(data.candies, data.extraCandies))
  );
}
